content-services/content-sources-backend
# Fixes 3930: speed up slow rpm searching

## Summary

Adds pg_trgm extension and new index on rpm names to speed up slow search rpm queries.

It looks like this solution makes the search rpms query about 10x faster (6ms vs 60ms locally) when there is a more selective filter on rpm name. 

Some caveats: 
- Index creation took a bit of time for me locally (fluctuated between 1 and 7 seconds)
- Index size is twice the size of the b-tree index that already exists on rpms.name
- Less selective filters on an rpm name (1 or 2 characters) are less efficient than the original search (70ms vs 60ms)

References:
https://www.postgresql.org/docs/current/pgtrgm.html#PGTRGM-INDEX
https://www.depesz.com/2011/02/19/waiting-for-9-1-faster-likeilike/

## Testing steps

- Run the migration to add the new extension and index (for QE, migration happens on deployment)
- Make a request to /rpms/names with a search term like `3proxy`, should be faster than before
- Verify introspection takes about the same time with or without the index (for me it was about a 1 ms difference)

## Checklist
